src/tiled/changepolygon.cpp

```diff
@@ -76,3 +76,50 @@ void TogglePolygonPolyline::toggle()
 {
     mMapObject->setShape((mMapObject->shape() == MapObject::Polygon) ? MapObject::Polyline : MapObject::Polygon);
 }
+
+SplitPolyline::SplitPolyline(MapDocument *mapDocument,
+                             MapObject *mapObject,
+                             int index)
+    : mMapDocument(mapDocument)
+    , mFirstPolyline(mapObject)
+    , mEdgeIndex(index)
+    , mObjectIndex(-1)
```

Comment:
Rather than putting the new polyline as the last object in the group, it would be nice to use the index of the polyline being split + 1.